### DISCUSSION TITLE
Is there a way to unhide entity property when the entity is serialized only for one time?

### DISCUSSION THREAD

**Discussion by ChristosTanailidis** [mikro-orm/mikro-orm#2913] [Q&A] [answered]
Hello. Let's say we have an entity with a property field which is set to ``` { hidden: true } ```. When we use ``` const newEntity em.create(Entity, { propertyField: 'something' } ```, is there a way to reveal that property only once during serialization?
It's like, i want to make an exclusion for hidden: true. I know it has something to do with the toJSON() method but i haven't managed to make it work.

**Comment by B4nan** [comment, marked as answer]
I don't follow, this flag is to hide things on serialization, if you dont want to have the property hidden during serialization, dont use it?

**Reply by AlexandrosMelissas** [reply]
Yeah but sometimes there are exceptions. For example, let's say the entity has a secret property field. When the entity is created, we want to return the secret field to the user only for that time. But generally, we want to keep the secret field hidden.

**Reply by B4nan** [reply]
You should handle that on that place where you need it, serialize the entity and set the value manually on the DTO. Or you could use some virtual property that would be set from hooks (e.g. propagate the secret in afterCreate hook).

**Reply by AlexandrosMelissas** [reply]
I see. I think i found a feature that might seem useful to him here https://mikro-orm.io/docs/defining-entities#lazy-scalar-properties . Lazy seems to be like "hidden" but more versatile since it can be fetched on choice.

**Reply by AlexandrosMelissas** [reply]
Although, lazy can't be accessed in runtime, but i dont think it matters in his case. He just wants to return it

**Reply by ChristosTanailidis** [reply]
I think that lazy property suits me most. 
Thanks a lot! :)

**Reply by SpencerKaiser** [reply]
@B4nan just voicing my opinion here too; I think the `hidden` property is incredible for things that are "private" but I have a similar concern to the above and don't really have a solid way of saying "by default this should be private, but there are times where I'd like to serialize this" (like when a user retrieves their entity vs another user wanting to see the "public" fields for a user)

Seems like `lazy` is an ok workaround, but the little note on `refresh` makes me a bit uneasy and I'd rather have to do hacky stuff to unhide something than risk something not being loaded at runtime.

In a perfect world, it'd be nice to have something similar to `.populated(bool)` on the property itself like `.hidden(bool)` where I could override whatever the default is for that property only for serialization purposes; thoughts on something like that?

**Reply by B4nan** [reply]
You can use explicit serialization if you want to have absolute control.

https://mikro-orm.io/docs/serializing#explicit-serialization

> but the little note on refresh makes me a bit uneas

With v5 you almost never need to refresh, reloading of entities now merges the diff, ignoring properties you already touched. Things are much smarter nowadays.

https://mikro-orm.io/blog/mikro-orm-5-released#automatic-refreshing-of-loaded-entities

Also, you can simply `em.populate()` lazy scalar props, no need to use `em.findOne()` at all.
